src/Functions/FunctionMapMapped.h

```diff
@@ -0,0 +1,198 @@
+        {
+            const auto & map_with_type_and_name = arguments[i];
+
+            ColumnPtr column_map_ptr = map_with_type_and_name.column;
+            const auto * column_map = checkAndGetColumn<ColumnMap>(column_map_ptr.get());
+
+            const DataTypePtr & map_type_ptr = map_with_type_and_name.type;
+            const auto * map_type = checkAndGetDataType<DataTypeMap>(map_type_ptr.get());
+
+            if (!column_map)
+            {
+                const ColumnConst * column_const_map = checkAndGetColumnConst<ColumnMap>(column_map_ptr.get());
+                if (!column_const_map)
+                    throw Exception("Expected map column, found " + column_map_ptr->getName(), ErrorCodes::ILLEGAL_COLUMN);
+                column_map_ptr = recursiveRemoveLowCardinality(column_const_map->convertToFullColumn());
+                column_map = checkAndGetColumn<ColumnMap>(column_map_ptr.get());
+            }
+
+            if (!map_type)
+                throw Exception("Expected map type, found " + map_type_ptr->getName(), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+            if (!offsets_column)
+            {
+                offsets_column = column_map->getNestedColumn().getOffsetsPtr();
+            }
+            else
+            {
+                /// The first condition is optimization: do not compare data if the pointers are equal.
+                if (column_map->getNestedColumn().getOffsetsPtr() != offsets_column
+                    && column_map->getNestedColumn().getOffsets() != typeid_cast<const ColumnArray::ColumnOffsets &>(*offsets_column).getData())
+                    throw Exception("maps passed to " + getName() + " must have equal size", ErrorCodes::SIZES_OF_ARRAYS_DOESNT_MATCH);
+            }
+
+            if (i == 1)
+            {
+                column_first_map_ptr = column_map_ptr;
+                column_first_map = column_map;
+            }
+
+            maps.emplace_back(ColumnWithTypeAndName(
+                column_map->getNestedData().getColumnPtr(0), recursiveRemoveLowCardinality(map_type->getKeyType()), map_with_type_and_name.name+".key"));
+            maps.emplace_back(ColumnWithTypeAndName(
+                column_map->getNestedData().getColumnPtr(1), recursiveRemoveLowCardinality(map_type->getValueType()), map_with_type_and_name.name+".value"));
+        }
+
+        /// Put all the necessary columns multiplied by the sizes of maps into the columns.
+        auto replicated_column_function_ptr = IColumn::mutate(column_function->replicate(column_first_map->getNestedColumn().getOffsets()));
+        auto * replicated_column_function = typeid_cast<ColumnFunction *>(replicated_column_function_ptr.get());
+        replicated_column_function->appendArguments(maps);
+
+        auto lambda_result = replicated_column_function->reduce().column;
+        if (lambda_result->lowCardinality())
+            lambda_result = lambda_result->convertToFullColumnIfLowCardinality();
```

Comment:
Can we just forbid lambdas returning low cardinality if we don't support this case?